### PR TITLE
Io

### PR DESCRIPTION
**Pull request by maxnoe**
This adds functionality originally implemented in the `classifier-tools` to read from different input formats into pandas data frames.

@mackaiver and I thought, this would fit well here, as this is not only needed in the classifier-tools but also in our day-to-day business.

### CODE REVIEW

File: fact/io.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    elif extension == '.json':
+        df.to_json(file_path)
+
+    elif extension == '.csv':
+        df.to_csv(file_path, delimiter=',', index=False)
+
+    else:
+        raise IOError(
+            'cannot write tabular data with format {}. Allowed formats: {}'.format(
+                extension, 'hdf5, json, csv'
```
Comment (kbruegge) [inline fact/io.py]: use the `allowed_extension` tuple here instead of a handwritten list?

File: fact/io.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                file_path,
+                key=key or 'events',
+                columns=columns,
+            )
+
+    elif extension == '.json':
+        with open(file_path, 'r') as j:
+            d = json.load(j)
+            df = pd.DataFrame(d)
+    elif extension in ('.jsonl', '.jsonlines'):
```
Comment (kbruegge) [inline fact/io.py:184]: we should add these extensions to the `allowed_extensions`  thingy.

File: fact/io.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+]
+
+log = logging.getLogger(__name__)
+
+
+allowed_extensions = ('.hdf', '.hdf5', '.h5', '.json', '.csv')
+native_byteorder = native_byteorder = {'little': '<', 'big': '>'}[sys.byteorder]
+
+
+def write_data(df, file_path, key='table'):
```
Comment (kbruegge) [inline fact/io.py]: Should we maybe also write 'non-pandas' hdf files? might be overkill eh?

**Review by kbruegge — APPROVED** [review]
Looks good mostly

File: fact/io.py (hunk 3 above)
Comment (maxnoe) [reply fact/io.py]: No, totally not. It's already implemented in erna. This PR is about moving stuff from klaas here. I will open another one moving the stuff from erna here.

File: fact/io.py (hunk 1 above)
Comment (maxnoe) [reply fact/io.py]: Jup.

File: fact/io.py (hunk 2 above)
Comment (maxnoe) [reply fact/io.py:184]: Yes.

Comment (maxnoe) [reply fact/io.py:184]: done

**Pull request merged**